include/mapbox/geometry/point.hpp

```diff
@@ -20,19 +23,19 @@ struct point
 };
 
 template <typename T>
-bool values_are_equal(T a, T b)
+inline bool values_are_equal(T a, T b)
 {
     return a == b;
 }
 
 template <>
-bool values_are_equal<double>(double a, double b)
+inline bool values_are_equal<double>(double a, double b)
 {
     return std::fabs(a - b) < (3.0 * std::numeric_limits<double>::epsilon());
 }
 
 template <>
-bool values_are_equal<float>(float a, float b)
+inline bool values_are_equal<float>(float a, float b)
 {
     return std::fabs(a - b) < (3.0 * std::numeric_limits<float>::epsilon());
```

Comment:
Q: where this magic number `3` came from ?
/cc @flippmoke

Reply:
I picked the magic value while using wagyu. However, I think we should change the way this works. This fails horribly when dealing with very large values where equality might be better dealt with in a better way.

Reply:
ok, thanks for clarification. The place to start is : http://www.boost.org/doc/libs/1_65_1/boost/test/tools/floating_point_comparison.hpp

Reply:
Ref: https://github.com/qt/qtbase/blob/5.11/src/corelib/global/qglobal.h#L821-L829